pymc_experimental/model/transforms/autoreparam.py

```diff
@@ -174,15 +176,18 @@ def vip_reparam_node(
 ) -> Tuple[ModelDeterministic, ModelNamed]:
     if not isinstance(node.op, RandomVariable | SymbolicRandomVariable):
         raise TypeError("Op should be RandomVariable type")
-    rv = node.default_output()
-    try:
-        [rv_shape] = constant_fold([rv.shape])
-    except NotConstantValueError:
-        raise ValueError("Size should be static for autoreparametrization.")
+    _, size, *_ = node.inputs
+    eval_size = size.eval()
+    if eval_size is not None:
+        rv_shape = tuple(eval_size)
+    else:
+        rv_shape = ()
+    lam_name = f"{name}::lam_logit__"
+    _log.debug(f"Creating {lam_name} with shape of {rv_shape}")
     logit_lam_ = pytensor.shared(
```

Comment:
Does this need to be a shared variable? How/who is controlling it?

Reply:
it will appear named variable in the model, like pm.Data

Reply:
Is there a better way to create new pm.Data?

Reply:
Or you can have a shared variable but update the shape later? Does your algorithm now the shape of the rv value variable? It could set the this shared variable then

Reply:
Okay maybe this is fine